### PR TITLE
[FLINK-13706][hive] add documentation of how to use Hive functions in…

### PR DESCRIPTION
**Pull request by bowenli86**
… Flink

## What is the purpose of the change

Add documentations of how to use Hive functions in Flink

## Brief change log

- Add documentations of how to use Hive functions in Flink

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by bowenli86** [comment]
cc @xuefuz @lirui-apache @zjuwangg @sjwiesman

File: docs/dev/table/hive/hive_functions.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+Supported UDF types include:
+
+- UDF
+- GenericUDF
+- GenericUDTF
+- UDAF
+- GenericUDAFResolver2
+
+Upon query planning and execution, Hive's UDF and GenericUDF are automatically translated into Flink's ScalarFunction, 
+Hive's GenericUDTF are automatically translated into Flink's TableFunction,
```
Comment (xuefuz) [inline docs/dev/table/hive/hive_functions.md]: are => is ?

File: docs/dev/table/hive/hive_functions.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+- GenericUDF
+- GenericUDTF
+- UDAF
+- GenericUDAFResolver2
+
+Upon query planning and execution, Hive's UDF and GenericUDF are automatically translated into Flink's ScalarFunction, 
+Hive's GenericUDTF are automatically translated into Flink's TableFunction, 
+and Hive's UDAF and GenericUDAFResolver2 are translated into Flink's AggregateFunction.
+
+To use Hive User Defined Functions, user must set a HiveCatalog backed by Hive Metastore that contains that function as the current catalog, and have to use Blink planner.
```
Comment (xuefuz) [inline docs/dev/table/hive/hive_functions.md]: Hive User Defined Functions => a Hive User Defined Function to be consistent with "that function"

as => in?

File: docs/dev/table/hive/hive_functions.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+Flink SQL> insert into mysinktable select a, s, sum(b), myudaf(b) from (%s) group by a, s;
+
+{% endhighlight %}
+
+
+### Limitations
+
+Hive built-in functions are currently not supported out of box in Flink. To use Hive built-in functions, users must register them manually in Hive Metastore first.
+
+Support for Hive functions have only been tested for Flink batch in Blink planner.
```
Comment (xuefuz) [inline docs/dev/table/hive/hive_functions.md]: have => has

File: docs/dev/table/hive/hive_functions.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	@Override
+	public void close() {
+	}
+}
+
+{% endhighlight %}
+
+Users can use them in SQL as:
```
Comment (xuefuz) [inline docs/dev/table/hive/hive_functions.md]: Maybe add some snippet from Hive cli to show those UDFs are registered in Hive.

File: docs/dev/table/hive/hive_functions.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+{% endhighlight %}
+
+
+### Limitations
+
+Hive built-in functions are currently not supported out of box in Flink. To use Hive built-in functions, users must register them manually in Hive Metastore first.
+
+Support for Hive functions have only been tested for Flink batch in Blink planner.
+
+Hive functions currently cannot be used across catalogs in Flink.
```
Comment (xuefuz) [inline docs/dev/table/hive/hive_functions.md]: Do we need to mention somewhere that to use those udfs, the relevant jars have to be in Flink's classpath?

Comment (zjuwangg) [reply docs/dev/table/hive/hive_functions.md]: Should we mention DataTypes support limitation here?

Comment (bowenli86) [reply docs/dev/table/hive/hive_functions.md]: added

**Comment by bowenli86** [comment]
cc @xuefuz @lirui-apache @zjuwangg

**Pull request closed without merging**